src/security.cs

```diff
@@ -337,178 +337,220 @@ interface KeysAccessible {
 		[Field ("kSecAttrCertificateType")]
 		IntPtr CertificateType { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrCertificateEncoding")]
 		IntPtr CertificateEncoding { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrKeyClass")]
 		IntPtr KeyClass { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrApplicationLabel")]
 		IntPtr ApplicationLabel { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrIsPermanent")]
 		IntPtr IsPermanent { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrIsSensitive")]
 		IntPtr IsSensitive { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrIsExtractable")]
 		IntPtr IsExtractable { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrApplicationTag")]
 		IntPtr ApplicationTag { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrKeyType")]
 		IntPtr KeyType { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrKeySizeInBits")]
 		IntPtr KeySizeInBits { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrEffectiveKeySize")]
 		IntPtr EffectiveKeySize { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrCanEncrypt")]
 		IntPtr CanEncrypt { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrCanDecrypt")]
 		IntPtr CanDecrypt { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrCanDerive")]
 		IntPtr CanDerive { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrCanSign")]
 		IntPtr CanSign { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrCanVerify")]
 		IntPtr CanVerify { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrCanWrap")]
 		IntPtr CanWrap { get; }
 
+		[iOS (2,0), Mac (10,6), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrCanUnwrap")]
 		IntPtr CanUnwrap { get; }
 
-		[iOS (9,0)]
-		[Mac (10,12)]
+		[iOS (9,0), Mac (10,12), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrTokenID")]
 		IntPtr TokenID { get; }
 
-		[iOS (9,0)]
-		[NoMac]
+		[iOS (9,0), Mac (10,12,1), TV (9,0), Watch (2,0)]
 		[Field ("kSecAttrTokenIDSecureEnclave")]
 		IntPtr SecureEnclave { get; }
 
-		[iOS (10,0)]
-		[Mac (10,12)]
-		[Watch (3,0)]
-		[TV (10,0)]
+		[iOS (10,0), Mac (10,12), TV (10,0), Watch (3,0)]
 		[Field ("kSecAttrAccessGroupToken")]
 		IntPtr AccessGroupToken { get; }
 
 		// note: kSecAttrPersistentReference (beta 1) is a typo that was not removed
-		[iOS (11,0)][TV (11,0)][Watch (4,0)][Mac (10,13)]
+		[iOS (11,0), Mac (10,13), TV (11,0), Watch (4,0)]
 		[Field ("kSecAttrPersistentReference")]
 		IntPtr PersistentReference { get; }
+
+		[iOS (2,0), Mac (10,8), TV (9,0), Watch (2,0)]
+		[Field ("kSecPrivateKeyAttrs")]
+		IntPtr PrivateKeyAttributes { get; }
```

Comment:
You would need to add the strongly typed API to `Items.cs`. See https://github.com/xamarin/xamarin-macios/blob/master/src/Security/Items.cs#L771-L782 as example.

Reply:
I respectfully disagree, as the SecRecord-class as fare as I can tell is used for querying and retrieving data from the KeyChain, while the kSecPrivateKeyAttrs and kSecPublicKeyAttrs is only used as part of the generation of new keys cf. https://developer.apple.com/documentation/security/kSecPrivateKeyAttrs
```
The corresponding value is a CFDictionary
containing key-value pairs for attributes
specific to the private key to be generated.
```

Therefore it is for me unclear what strongly typed API in `Items.cs` would look like. If anything I would be more appropiate to have it in `SecCertificate.cs` as this is where the key generation is done.

Reply:
I think you're right that the SecRecord-class is probably not the right place to put the strongly typed APIs for `kSecPrivateKeyAttrs` and `kSecPublicKeyAttrs`.

However we should expose them somewhere (but this will need to be revisited).